### PR TITLE
Search implementation for v5, v6, v7

### PR DESCRIPTION
**Pull request by parikshitg**
### Description

* Includes search implementation for Elasticsearch v5, v6, v7. 
* Since v5 and v6 do not support sorting using `_seq_no`, we can now specify the `sortby` field in the config. Default sorting will be on `_seq_no` for v7 and v8.
* Added retry logic for search api.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-elasticsearch/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,7 +50,8 @@ ElasticSearch source connector allows you to move data from multiple Elasticsear
...
 | `certificateFingerprint` | [v: 7, 8] SHA256 hex fingerprint given by Elasticsearch on first launch.                                                                                                                                                                         | `false`                                              |          |
-| `indexes`                  | The name of the indexes to read the data from.                                                                                                                                                                                                      | `true`                                               |          |
+| `indexes.*.sortBy`                  | The sortby field for each index to be used by elasticsearch search api.(A field must be specified for v5, v6 as it does not support sorting using the default `_seq_no`)                                                                                                                                                                                                       | `true`                                               |    `_seq_no`      |
```
Comment (raulb) [inline README.md:53]: This field shouldn't be required when it has a default value.

```suggestion
| `indexes.*.sortBy`                  | The `sortby` field for each index to be used by elasticsearch search api. A field must be specified for v5, v6 as it does not support sorting using the default `_seq_no`.                                                                                                                                                                                                       | `false`                                               |    `_seq_no`      |
```

File: README.md [hunk 2]
```diff
@@ -50,7 +50,8 @@ ElasticSearch source connector allows you to move data from multiple Elasticsear
...
+| `indexes.*.sortOrder`                  | The sortOrder (asc or desc) for each index to be used by elasticsearch search api.                                                                                                                                                                                                      | `true`                                               |   `asc`       |
```
Comment (raulb) [inline README.md:54]: Same as above, this field shouldn't be required when providing a default value.

```suggestion
| `indexes.*.sortOrder`                  | The sortOrder (asc or desc) for each index to be used by elasticsearch search api.                                                                                                                                                                                                      | `true`                                               |   `asc`       |
```

File: source/config.go [hunk 3]
```diff
@@ -39,12 +39,21 @@ type Config struct {
...
 	// The number of items stored in bulk in the index. The minimum value is `1`, maximum value is `10000`.
 	BatchSize int `json:"batchSize" default:"1000"`
 	// This period is used by workers to poll for new data at regular intervals.
 	PollingPeriod time.Duration `json:"pollingPeriod" default:"5s"`
+	// The maximum number of retries of failed operations.
+	Retries int `json:"retries" default:"0"`
+}
+
+type Sort struct {
+	// The sortbyField for each index to be used by elasticsearch search api.
```
Comment (raulb) [inline source/config.go:53]:
```suggestion
	// The sortby field for each index to be used by elasticsearch search api.
```

File: source/config.go [hunk 4]
```diff
@@ -39,12 +39,21 @@ type Config struct {
...
+	SortBy string `json:"sortBy" default:"_seq_no"`
+	// The sortOrders(asc or desc) for each index to be used by elasticsearch search api.
```
Comment (raulb) [inline source/config.go:55]:
```suggestion
	// The sortOrder(asc or desc) for each index to be used by elasticsearch search api.
```

File: source/paramgen.go [hunk 5]
```diff
@@ -75,6 +81,12 @@ func (Config) Parameters() map[string]config.Parameter {
 			Type:        config.ParameterTypeDuration,
 			Validations: []config.Validation{},
 		},
+		ConfigRetries: {
+			Default:     "0",
```
Comment (raulb) [inline source/paramgen.go:85]: Readme says default value is `1000`. We need to consolidate. Either, the code in config or readme.

**Review by raulb — CHANGES_REQUESTED** [review]
Thank you @parikshitg .

Left some comments inline with the code, and here are some other considerations:

- golangci-lint needs to be throwing some errors that need to be fixed.
- Need to rebase with main since this branch since out of date.
- The content of v5/search.go and v6/search.go seem to be identical. I’d suggest reusing the same code. Maybe also reutilize what’s possible for v7/search.go.

File: source/paramgen.go (hunk 5 above)
Comment (parikshitg) [reply source/paramgen.go:85]: updated.

File: source/config.go (hunk 4 above)
Comment (parikshitg) [reply source/config.go]: updated.

File: source/config.go (hunk 3 above)
Comment (parikshitg) [reply source/config.go]: updated.

File: README.md (hunk 2 above)
Comment (parikshitg) [reply README.md]: updated.

File: README.md (hunk 1 above)
Comment (parikshitg) [reply README.md]: updated.

**Comment by parikshitg** [comment]
> Thank you @parikshitg .
> 
> Left some comments inline with the code, and here are some other considerations:
> 
> * golangci-lint needs to be throwing some errors that need to be fixed.
> * Need to rebase with main since this branch since out of date.
> * The content of v5/search.go and v6/search.go seem to be identical. I’d suggest reusing the same code. Maybe also reutilize what’s possible for v7/search.go.

Thanks for your review! I have made the changes. 
* There is a common function that creates a search body. 
* Search() for each version now has the minimal code for that client version.

**Review by raulb — APPROVED** [review]

**Pull request merged**